### ISSUE TITLE
Error when trying to purge Blitz cache

### ISSUE THREAD

**Issue by danethomas** [putyourlightson/craft-blitz-cloudfront#9]
Here's the error I'm seeing when trying to purge Blitz cache and running `craft queue/listen --verbose`

```
2022-11-08 19:38:14 [644511] Purging Blitz cache (attempt: 2, pid: 773) - Started
2022-11-08 19:38:14 [644511] Purging Blitz cache (attempt: 2, pid: 773) - Error (0.012 s)
> yii\base\ErrorException: Attempt to read property "uri" on array
Stack trace:
#0 /usr/share/nginx/html/craft/vendor/putyourlightson/craft-blitz-cloudfront/src/CloudFrontPurger.php(160): yii\base\ErrorHandler->handleError()
#1 [internal function]: putyourlightson\blitzcloudfront\CloudFrontPurger->putyourlightson\blitzcloudfront\{closure}()
#2 /usr/share/nginx/html/craft/vendor/putyourlightson/craft-blitz-cloudfront/src/CloudFrontPurger.php(159): array_map()
#3 [internal function]: putyourlightson\blitzcloudfront\CloudFrontPurger->purgeUrisWithProgress()
#4 /usr/share/nginx/html/craft/vendor/putyourlightson/craft-blitz/src/jobs/DriverJob.php(69): call_user_func()
#5 /usr/share/nginx/html/craft/vendor/yiisoft/yii2-queue/src/Queue.php(246): putyourlightson\blitz\jobs\DriverJob->execute()
#6 /usr/share/nginx/html/craft/vendor/yiisoft/yii2-queue/src/cli/Queue.php(162): yii\queue\Queue->handleMessage()
#7 /usr/share/nginx/html/craft/vendor/yiisoft/yii2-queue/src/cli/Command.php(146): yii\queue\cli\Queue->execute()
#8 [internal function]: yii\queue\cli\Command->actionExec()
#9 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()
#10 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Controller.php(178): yii\base\InlineAction->runWithParams()
#11 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/console/Controller.php(180): yii\base\Controller->runAction()
#12 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Module.php(552): yii\console\Controller->runAction()
#13 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/console/Application.php(180): yii\base\Module->runAction()
#14 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/console/Application.php(90): yii\console\Application->runAction()
#15 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/console/Application.php(147): craft\console\Application->runAction()
#16 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/console/Application.php(121): yii\console\Application->handleRequest()
#17 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Application.php(384): craft\console\Application->handleRequest()
#18 /usr/share/nginx/html/craft/craft(13): yii\base\Application->run()
#19 {main}
2022-11-08 08:38:14 [queue.ERROR] [craft\queue\QueueLogBehavior::afterError]  [644511] Purging Blitz cache (attempt: 2, pid: 773) - Error (time: 0.012s): Attempt to read property "uri" on array {"memory":33060096}
2022-11-08 08:38:14 [queue.ERROR] [yii\base\ErrorException:2] Attempt to read property "uri" on array {"trace":["#0 /usr/share/nginx/html/craft/vendor/putyourlightson/craft-blitz-cloudfront/src/CloudFrontPurger.php(160): yii\\base\\ErrorHandler->handleError()","#1 [internal function]: putyourlightson\\blitzcloudfront\\CloudFrontPurger->putyourlightson\\blitzcloudfront\\{closure}()","#2 /usr/share/nginx/html/craft/vendor/putyourlightson/craft-blitz-cloudfront/src/CloudFrontPurger.php(159): array_map()","#3 [internal function]: putyourlightson\\blitzcloudfront\\CloudFrontPurger->purgeUrisWithProgress()","#4 /usr/share/nginx/html/craft/vendor/putyourlightson/craft-blitz/src/jobs/DriverJob.php(69): call_user_func()","#5 /usr/share/nginx/html/craft/vendor/yiisoft/yii2-queue/src/Queue.php(246): putyourlightson\\blitz\\jobs\\DriverJob->execute()","#6 /usr/share/nginx/html/craft/vendor/yiisoft/yii2-queue/src/cli/Queue.php(162): yii\\queue\\Queue->handleMessage()","#7 /usr/share/nginx/html/craft/vendor/yiisoft/yii2-queue/src/cli/Command.php(146): yii\\queue\\cli\\Queue->execute()","#8 [internal function]: yii\\queue\\cli\\Command->actionExec()","#9 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/InlineAction.php(57): call_user_func_array()","#10 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Controller.php(178): yii\\base\\InlineAction->runWithParams()","#11 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/console/Controller.php(180): yii\\base\\Controller->runAction()","#12 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Module.php(552): yii\\console\\Controller->runAction()","#13 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/console/Application.php(180): yii\\base\\Module->runAction()","#14 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/console/Application.php(90): yii\\console\\Application->runAction()","#15 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/console/Application.php(147): craft\\console\\Application->runAction()","#16 /usr/share/nginx/html/craft/vendor/craftcms/cms/src/console/Application.php(121): yii\\console\\Application->handleRequest()","#17 /usr/share/nginx/html/craft/vendor/yiisoft/yii2/base/Application.php(384): craft\\console\\Application->handleRequest()","#18 /usr/share/nginx/html/craft/craft(13): yii\\base\\Application->run()","#19 {main}"],"memory":33060528,"exception":"[object] (yii\\base\\ErrorException(code: 2): Attempt to read property \"uri\" on array at /usr/share/nginx/html/craft/vendor/putyourlightson/craft-blitz-cloudfront/src/CloudFrontPurger.php:160)"}
```

**System**

- "craftcms/cms": "4.3.1",
- "putyourlightson/craft-blitz": "4.2.3",
- "putyourlightson/craft-blitz-cloudfront": "^3.0", (installed version from .lock is 3.02)

**Comment by bencroker** [comment]
How exactly are you kicking off the purge cache job?

**Comment by danethomas** [comment]
It was just being triggered by saving an entry. I haven't tried a full blitz cache flush yet, can do that today.

I just ran the queue manually as I noticed the blitz cache jobs were backing up in the control panel.

**Comment by bencroker** [comment]
This appears to be caused by custom site URIs not being converted to models. 

Fixed in https://github.com/putyourlightson/craft-blitz/commit/33a37cabdf7ea37cee4409799049a7792ad7a322 for the next release. If you;d like to try this then you can do so by changing the `putyourlightson/craft-blitz` requirement in `composer.json` to `"4.3.0.x-dev as 4.3.0"` and running `composer update`.

**Comment by bencroker** [comment]
I ended up tracking this down to a deeper issue, fixed in https://github.com/putyourlightson/craft-blitz-cloudfront/commit/b4e7f65b2e7fa05b20095ab041fcf27614a42794 and released in 3.0.3 (updating Blitz itself is not required). Thanks for your help in troubleshooting!
